src/main/java/org/opensearch/security/hasher/PasswordHasherFactory.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+package org.opensearch.security.hasher;
+
+import org.opensearch.common.settings.Settings;
+import org.opensearch.security.support.ConfigConstants;
+
+import static org.opensearch.security.support.ConfigConstants.BCRYPT;
+import static org.opensearch.security.support.ConfigConstants.PBKDF2;
+
+public class PasswordHasherFactory {
+
+    public static PasswordHasher createPasswordHasher(Settings settings) {
+
+        String algorithm = settings.get(
+            ConfigConstants.SECURITY_PASSWORD_HASHING_ALGORITHM,
+            ConfigConstants.SECURITY_PASSWORD_HASHING_ALGORITHM_DEFAULT
+        );
+
+        PasswordHasher passwordHasher;
+        switch (algorithm.toLowerCase()) {
+            case BCRYPT:
+                passwordHasher = getBCryptHasher(settings);
+                break;
+            case PBKDF2:
+                passwordHasher = getPBKDF2Hasher(settings);
+                break;
+            default:
+                throw new IllegalArgumentException("Password hashing algorithm not supported");
+        }
+        return passwordHasher;
+    }
+
+    private static PasswordHasher getBCryptHasher(Settings settings) {
+        int rounds = settings.getAsInt(
+            ConfigConstants.SECURITY_PASSWORD_HASHING_BCRYPT_ROUNDS,
+            ConfigConstants.SECURITY_PASSWORD_HASHING_BCRYPT_ROUNDS_DEFAULT
+        );
+        String minor = settings.get(
+            ConfigConstants.SECURITY_PASSWORD_HASHING_BCRYPT_MINOR,
+            ConfigConstants.SECURITY_PASSWORD_HASHING_BCRYPT_MINOR_DEFAULT
+        ).toUpperCase();
+
+        if (rounds < 4 || rounds > 31) {
+            throw new IllegalArgumentException("BCrypt rounds must be between 4 and 31.");
+        }
+        if (!minor.equals("A") && !minor.equals("B") && !minor.equals("Y")) {
+            throw new IllegalArgumentException("BCrypt minor must be 'A', 'B', or 'Y'.");
+        }
```

Comment:
can we make it more readable 

```
Set<String> ALLOWED_BCRYPT_MINORS = Set.of("A", "B", "Y");

if (!ALLOWED_BCRYPT_MINORS.contains(minor)) {
    throw new IllegalArgumentException("BCrypt minor must be 'A', 'B', or 'Y'.");
}

or 

```
if (!Set.of("A", "B", "Y").contains(minor)) {
    throw new IllegalArgumentException("BCrypt minor must be 'A', 'B', or 'Y'.");
}

```

```

Reply:
Password4j will throw their own BadParametersException (implements IllegalArgumentException). Personally, i'd leave the validation here so we don't expose Password4j exceptions needlessly